eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Ledger.Conway.Rules.Ratify (
+  RatifyState (..),
+  RatifyEnv (..),
+  RatifySignal (..),
+) where
+
+import Cardano.Ledger.BaseTypes (ShelleyBase)
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Conway.Era (ConwayENACT, ConwayRATIFY)
+import Cardano.Ledger.Conway.Governance (
+  EraGovernance,
+  GovernanceAction,
+  GovernanceActionId,
+  GovernanceActionState (..),
+  RatifyState (..),
+  VoteDecision (..),
+  VoterRole (..),
+ )
+import Cardano.Ledger.Conway.Rules.Enact (EnactState (..))
+import Cardano.Ledger.Core
+import Cardano.Ledger.Credential (Credential)
+import Cardano.Ledger.Keys (HasKeyRole (..), KeyRole (..))
+import Cardano.Ledger.Slot (EpochNo (..))
+import Control.State.Transition.Extended (Embed (..), STS (..), TRC (..), TransitionRule, judgmentContext, trans)
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import Data.Ratio ((%))
+import Data.Sequence.Strict (StrictSeq (..))
+import qualified Data.Set as Set
+import Data.Void (Void, absurd)
+
+data RatifyEnv era = RatifyEnv
+  { reStakeDistr :: !(Map (Credential 'Staking (EraCrypto era)) Coin)
+  , reCurrentEpoch :: !EpochNo
+  , reRoles :: !(Map (Credential 'Voting (EraCrypto era)) VoterRole)
+  }
+
+newtype RatifySignal era
+  = RatifySignal
+      ( StrictSeq
+          ( (GovernanceActionId (EraCrypto era))
+          , (GovernanceActionState era)
+          )
+      )
+
+instance
+  ( Era era
+  , Embed (EraRule "ENACT" era) (ConwayRATIFY era)
+  , State (EraRule "ENACT" era) ~ EnactState era
+  , Environment (EraRule "ENACT" era) ~ ()
+  , Signal (EraRule "ENACT" era) ~ GovernanceAction era
+  ) =>
+  STS (ConwayRATIFY era)
+  where
+  type Environment (ConwayRATIFY era) = RatifyEnv era
+  type PredicateFailure (ConwayRATIFY era) = Void
+  type Signal (ConwayRATIFY era) = RatifySignal era
+  type State (ConwayRATIFY era) = RatifyState era
+  type BaseM (ConwayRATIFY era) = ShelleyBase
+
+  initialRules = []
+  transitionRules = [ratifyTransition]
+
+--- Constants
```

Comment:
I guess we are just hard-coding these temporarily, until we add them to the thresholds?